### PR TITLE
Exclude NaNs from map FITS headers

### PR DESCRIPTION
**Pull request by dstansby**
Fixes https://github.com/sunpy/sunpy/issues/3819 - seems reasonable to raise a warning here similar to the other warnings for non-valid FITS keywords.

### CODE REVIEW

**Comment by nabobalis** [comment]
Stupid question, on the issue you said that the warning has the reason:

```
/Users/dstansby/github/sunpy/sunpy/map/mapbase.py:443: UserWarning: Unable to treat `.meta` as a FITS header, assuming a simple WCS. The exception raised was:
Floating point nan values are not allowed in FITS headers.
  warnings.warn("Unable to treat `.meta` as a FITS header, assuming a simple WCS. "
```
Do we need a second warning?

**Comment by kakirastern** [comment]
Obviously not

**Comment by kakirastern** [comment]
Needs be just combine them into the one warning ⚠️

**Comment by dstansby** [comment]
With this PR a different warning is raised, and the `Unable to treat `.meta` as a FITS header, assuming a simple WCS.` isn't raised, since the NaN values get filtered out and `.meta` can be treated as a FITS header.

File: sunpy/io/fits.py [hunk 1]
```diff
@@ -23,6 +23,7 @@
    invalid header tags will result in an error so verifying it early on
    makes the header easier to work with later.
 """
+import math
```
Comment (nabobalis) [inline sunpy/io/fits.py:26]: Just need to move this to between sys and warnings.

**Review by Cadair — APPROVED** [review]

**Comment by Cadair** [comment]
I only have one concern with this, and it's really a question of if this should get backported or not.

What happens if we try and save one of these files with nan's in the header back to FITS, does astropy already strip them? If it doesn't then we might want to consider not backporting this as we would be changing the headers we save to disk on people.

**Comment by kakirastern** [comment]
Should we at least get the code style check passing too, only change is a one-liner after all with the `import math` statement...

**Review by kakirastern — APPROVED** [review]

**Comment by dstansby** [comment]
Yeah, I need to fix the tests, and add a new test (sorry, I should have left this in draft).

File: sunpy/io/fits.py (hunk 1 above)
Comment (dstansby) [reply sunpy/io/fits.py:26]: That's not how the alphabet works 🤣

**Comment by kakirastern** [comment]
Sounds good, just in general want our build to be passing on GitHub...

**Review by nabobalis — APPROVED** [review]

**Pull request merged**